docs/numba.md

```diff
@@ -0,0 +1,150 @@
+---
+jupytext:
+  text_representation:
+    extension: .md
+    format_name: myst
+    format_version: 0.12
+    jupytext_version: 1.9.1
+kernelspec:
+  display_name: Python 3
+  language: python
+  name: python3
+---
+
+```{currentmodule} tskit.jit.numba
+```
+
+(sec_numba)=
+
+# Numba Integration
+
+The `tskit.jit.numba` module provides classes for working with tree sequences
+from [Numba](https://numba.pydata.org/) jit-compiled Python code. Such code can run
+upto hundreds of times faster than normal Python, yet avoids the difficulties of writing
+C or other low-level code.
+
+:::{note}
+Numba is not a direct dependency of tskit, so will not be avaliable unless installed:
+
+```bash
+pip install numba
+```
+
+or
+
+```bash
+conda install numba
+```
+:::
+
+## Overview
+
+The numba integration provides:
+
+- **{class}`NumbaTreeSequence`**: A Numba-compatible representation of tree sequence data
+- **{class}`NumbaTreePosition`**: An class for efficient tree traversal
+- **{class}`NumbaEdgeRange`**: Container class for edge ranges during traversal
+
+These classes are designed to work within Numba's `@njit` decorated functions,
+allowing you to write high-performance tree sequence analysis code.
+
+## Basic Usage
+
+The ``tskit.jit.numba`` module is not imported with normal `tskit` so must be imported explicitly:
+```{code-cell} python
+import tskit
+import tskit.jit.numba as tskit_numba
+```
+
+Normal third-party classes such as {class}`tskit.TreeSequence` can't be used in `numba.njit` compiled
+functions so the {class}`tskit.TreeSequence` must be wrapped in a {class}`NumbaTreeSequence` by 
+{meth}`numba_tree_sequence`. This must be done outside `njit` code:
+
+```{code-cell} python
+import msprime
+
+ts = msprime.sim_ancestry(
+    samples=50000,
+    sequence_length=100000,
+    recombination_rate=0.1,
+    random_seed=42
+)
+numba_ts = tskit_numba.numba_tree_sequence(ts)
+print(type(numba_ts))
+```
+
+## Tree Traversal
```

Comment:
I normally think of "tree traversal" as iterating through the tree structure itself. Do you mean "Iterating through trees" here? I can't see any pre/postorder traversal code here.

Reply:
Good point, I was avoiding the word "iteration" not not confuse it with a Python iterator - but I'll change it back as this is more confusing!